question28/js/forgotpassword.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+import { validationInfo } from "./validation";
+import { Chance } from "chance";
+const chance = new Chance();
+
+const submitButton = document.getElementById('js-submit-button');
+const userInfoInputArea = document.getElementById('name_mail');
+const inputArea = document.getElementById('name_mail');
+
+const isValidStatus = {
+  name_mail: false,
+}
+
+const setValidationEvents = (isValid, targetForm) =>
+  isValid ? validEvent(targetForm) : invalidEvent(targetForm);
+
+const validEvent = (targetForm) => {
+  targetForm.classList.add("valid");
+  targetForm.classList.remove("invalid");
+  targetForm.nextElementSibling.textContent = "";
+  isValidStatus[targetForm.id] = true;
+};
+
+const invalidEvent = (targetForm) => {
+  targetForm.classList.add("invalid");
+  targetForm.classList.remove("valid");
+  targetForm.nextElementSibling.textContent = validationInfo[targetForm.id].errorMessage;
+  isValidStatus[targetForm.id] = false;
+};
+
+const validateInputValue = (e) => {
+  const targetForm = e.target;
+  const value = targetForm.value.trim();
+  const result = validationInfo[targetForm.id].validation(value);
+  setValidationEvents(result,targetForm);
+  toggleDisabledOfSubmitButton(checkAllValidity());
+} 
+
+const checkAllValidity = () => {
+  return Object.values(isValidStatus).every((result) => result );
+}
+
+const toggleDisabledOfSubmitButton = (isValid) => {
+  submitButton.disabled = isValid ? false : true;
+} 
+
+userInfoInputArea.addEventListener('blur', validateInputValue);
+
+const init = () => {
+  const inputData = inputArea.value;
+  const registeredJsonUserData = localStorage.getItem('userData'); 
+  const registeredUserData = JSON.parse(registeredJsonUserData); 
+  const tokenForPasswordReset = chance.apple_token();
+  const passwordResetPageUrl = "./register/password.html";
+
+  if (registeredJsonUserData == null) {
+    window.location.href = './notautherize.html';
+    return; 
+  }
+  if (registeredUserData.user_name == inputData || registeredUserData.user_email == inputData) {
+    window.location.href = `${passwordResetPageUrl}?token=${tokenForPasswordReset}`;
+    return; 
+  }
+  window.location.href = './notautherize.html';
+}
```

Comment:
【Add】
①If there is no information in local storage
→Go to ``./notautherize.html``

②If there is registration information in local storage, but the input values are incorrect.
→Go to ``./notautherize.html``

③If there is registration information in the local storage, and if the input information matches the registration information
→Go to ``./register/password.html``